src/test/crux-gateway/test-entity.ts

```diff
@@ -41,6 +43,30 @@ describe('CRUX Gateway Entity Tests', function() {
             });
             user1GatewayForUser2.sendMessage(testmsg);
         });
+
+        it('Test Certificate Creation', async function() {
```

Comment:
Look at test cases in test/crux-wallet-client.ts, make your assertions and all like those ones.

Reply:
Also add negative test cases for the `verify` function. In which cases should it return false? Simulate those cases

Reply:
> Look at test cases in test/crux-wallet-client.ts, make your assertions and all like those ones.

changed accordingly in updated PR

Reply:
> Also add negative test cases for the `verify` function. In which cases should it return false? Simulate those cases

Added few in updated PR. Will add more robust in another updated PR